xilem/examples/gallery_label_xilem_view.rs

```diff
@@ -0,0 +1,348 @@
+        } else if p == "l4i4" {
+            self.l4i4 = text_align_cycle(&self.l4i4);
+        } else if p == "l4i5" {
+            self.l4i5 = text_align_cycle(&self.l4i5);
+        } else if p == "l4i6" {
+            self.l4i6 = text_align_cycle(&self.l4i6);
+        } else if p == "l4i1x" {
+            self.l4i1x = text_x_align_cycle(&self.l4i1x);
+        } else {
+            warn!("expecting either of ‘l1–4i1–6’ or ‘l4i1x’, got {:?}", pos);
+        }
+    }
+}
+
+fn title_prose(text: impl Into<ArcStr>) -> Prose {
+    prose(text)
+        .text_size(18.0)
+        .alignment(TextAlignment::Justified)
+        .brush(css::GREEN)
+}
+fn txt_prose(text: impl Into<ArcStr>) -> Prose {
+    prose(text)
+        .text_size(14.0)
+        .alignment(TextAlignment::Start)
+        .brush(Color::from_rgb8(0x11, 0x11, 0x11))
+}
+fn lc(text: impl Into<ArcStr>) -> Label {
+    //colored label
+    label(text).brush(LABEL_COLOR)
+}
+fn app_logic(d: &mut AppState) -> impl WidgetView<AppState> + use<> {
+    let m_c = Color::from_rgb8(0x11, 0x11, 0x11); //main text
+    let l_c = LABEL_COLOR;
+    let mut i = 0;
+
+    portal(
+    flex((
+    (txt_prose("Xilem view::Label formats vGit@25-04 #8c25fea (in a ↕-scrollable area)").text_size(18.0),
+    if cfg!(debug_assertions) {txt_prose(
+     "This is a debug build, so you can use github.com/linebender/xilem/tree/main/masonry#debugging-features:
+        • F11 to toggle a rudimentary widget inspector
+        • F12 to toggle paint widget layout rectangles")
+    } else {txt_prose("This is not a debug build, so F11 widget inspector and F12 widget rectangles tools are not available)\ngithub.com/linebender/xilem/tree/main/masonry#debugging-features")},
+    label(format!("Label: Serif Bold 14 {LABEL_COLOR:?}")).text_size(14.0).weight(FontWeight::BOLD) // float bold=700, FontWeight::parse("normal") for css
+        .font(fontique::GenericFamily::Serif)
+        .alignment(TextAlignment::Start)
+        .brush(l_c) //impl Into<peniko:Brush> brush sets text color, but gradients and images are also supported Solid(color::AlphaColor<Srgb>) Gradient(Gradient) Image(Image),
+        .line_break_mode(LineBreaking::Overflow) //WordWrap Clip Overflow
+        ,
+    {i+=1;},title_prose(format!("§ {i} .alignment")),
+    txt_prose("  6 options: ≝Start Middle End Justified Left Right\n  https://docs.rs/parley/latest/parley/layout/enum.Alignment.html")
+    ),
+    (label("  • flex in a 200×220 sized_box to show the impact of constraints (buttons to change alignment)").alignment(TextAlignment::Justified).brush(m_c),
+    flex((
+      button("Δ1/6", |da:&mut AppState|{da.realign("l1i1");}),
+      button("Δ2/6", |da:&mut AppState|{da.realign("l1i2");}),
+      button("Δ3/6", |da:&mut AppState|{da.realign("l1i3");}),
+      button("Δ4/6", |da:&mut AppState|{da.realign("l1i4");}),
+      button("Δ5/6", |da:&mut AppState|{da.realign("l1i5");}),
+      button("Δ6/6", |da:&mut AppState|{da.realign("l1i6");}),
```

Comment:
It's not all that clear what these buttons are useful for?

Even so, with having them, I think they could probably also be changed into an enumarted loop over a sub-array of the state.

Reply:
For playing with various styling in real time. While currently for some styles all the variants are already included, so buttons are of limited use when you want to see e.g. the whole "table" of labels aligned the same way, that's not true for all styles - Cross-alignment doesn't list all the combos. Neither would font families (though those don't have dynamic buttons/pickers at the moment) and colors/sizes